src/algebra/ring/basic.lean

```diff
@@ -923,6 +929,24 @@ protected def function.surjective.ring
 { .. hf.add_comm_group f zero add neg sub nsmul zsmul,
   .. hf.monoid f one mul npow, .. hf.distrib f add mul }
 
+@[simp]
+lemma odd_add_one_iff_even {R : Type*} [ring R] (a : R) : odd (a + 1) ↔ even a :=
+by simp [even_iff_bit0, odd_iff_bit1, bit0, bit1]
+
+@[simp]
+lemma odd_sub_one_iff_even {R : Type*} [ring R] (a : R) : odd (a - 1) ↔ even a :=
+begin
+  refine ⟨_, _⟩,
+  { rintro ⟨w, hw⟩,
+    refine ⟨w + 1, _⟩,
+    rw [sub_eq_iff_eq_add.mp hw, mul_add, add_assoc, mul_one],
+    refl },
+  { rintro ⟨w, rfl⟩,
+    refine ⟨w - 1, _⟩,
+    rw [mul_sub, sub_add, sub_right_inj, mul_one, sub_eq_iff_eq_add.mpr],
+    refl }
+end
```

Comment:
Can't you do something like
```
  have h : a + 1 = (a - 1) + 2, by ring
  rw [← odd_add_one_iff_even, h],
  simp -- I hope this works, maybe you need a little bit of work?
```

Reply:
I had a version with ring, but it is not imported in this file...  Should I import wherever ring appears just for this proof?

Reply:
I had a version with ring, but it is not imported in this file... Should I import wherever ring appears just for this proof?

Reply:
No, in that case prove `h` by hand. That shouldn't be too hard.